src/data/set/intervals/monotone.lean

```diff
@@ -177,3 +178,81 @@ calc (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :
 ... = Ioi a ∩ Iio b : congr_arg2 (∩) ha.Union_Ioi_eq hb.Union_Iio_eq
 
 end Union
+
+section succ_order
+
+open order
+
+variables {α β : Type*} [partial_order α]
+
+lemma strict_mono_on.Iic_id_le [succ_order α] [is_succ_archimedean α] [order_bot α]
+  {n : α} {φ : α → α} (hφ : strict_mono_on φ (set.Iic n)) :
+  ∀ m ≤ n, m ≤ φ m :=
+begin
+  revert hφ,
+  refine succ.rec_bot (λ n, strict_mono_on φ (set.Iic n) → ∀ m ≤ n, m ≤ φ m)
+    (λ _ _ hm, hm.trans bot_le) _ _,
+  rintro k ih hφ m hm,
+  by_cases hk : is_max k,
+  { rw succ_eq_iff_is_max.2 hk at hm,
+    exact ih (hφ.mono $ Iic_subset_Iic.2 (le_succ _)) _ hm },
+  obtain (rfl | h) := le_succ_iff_eq_or_le.1 hm,
+  { specialize ih (strict_mono_on.mono hφ (λ x hx, le_trans hx (le_succ _))) k le_rfl,
+    refine le_trans (succ_mono ih) (succ_le_of_lt (hφ (le_succ _) le_rfl _)),
+    rw lt_succ_iff_eq_or_lt_of_not_is_max hk,
+    exact or.inl rfl },
+  { exact ih (strict_mono_on.mono hφ (λ x hx, le_trans hx (le_succ _))) _ h }
+end
+
+lemma strict_mono_on.Iic_le_id [pred_order α] [is_pred_archimedean α] [order_top α]
+  {n : α} {φ : α → α} (hφ : strict_mono_on φ (set.Ici n)) :
+  ∀ m, n ≤ m → φ m ≤ m :=
+@strict_mono_on.Iic_id_le αᵒᵈ _ _ _ _ _ _ (λ i hi j hj hij, hφ hj hi hij)
+
+variables [preorder β] {ψ : α → β}
+
+/-- A function `ψ` on a `succ_order` is strictly monotone before some `n` if for all `m` such that
+`succ m ≤ n`, we have `ψ m < ψ (succ m)`.
+
+We note that in the lemma `succ m ≤ n` and `m < n` are not equivalent since this is in general
+not true if we do not assume `no_max_order`. -/
+lemma strict_mono_on_Iic_of_lt_succ [succ_order α] [is_succ_archimedean α]
+  {n : α} (hψ : ∀ m, succ m ≤ n → ψ m < ψ (succ m)) :
```

Comment:
If in the assumption `hψ` you replace `succ m ≤ n` with `m < n`, this is a weaker assumption, right (i.e., easier to check). Isn't the lemma still true under this weaker assumption?